### ISSUE TITLE
Can't type left bracket in switch-buffer

### ISSUE THREAD

**Issue by glacambre** [atlas-engineer/next#313]
Hi. First, thank you for building and maintaining next, I feel like it is a very promising project :).

I built next from bdeb6347e92d3f65d486bba521a43f36811bf03c . When using `switch-buffer`, pressing `[` results in errors being thrown. It looks like next tries to build regexes from this string without escaping the characters. I tried to track down the cause of this bug but couldn't find the exact piece of code responsible for it.

Here's the error log:
```
[1] 28528
 <INFO> [18:23:38] next base.lisp (start) - NEXT::+VERSION+: "1.3.0-bdeb634"
 <INFO> [18:23:38] next base.lisp (load-lisp-file form-fun-4) -
  Loading configuration from /home/me/.dotfiles/default/.config/next/init.lisp...
 <WARN> [18:23:38] next minibuffer.lisp (echo) -
  Can't echo 'MINIBUFFER-MODE enabled.' without minibuffer or interface
> <INFO> [18:23:38] next remote.lisp (initialize-instance :after remote-interface) -
  Bus connection name: :1.4976
 <INFO> [18:23:38] next port.lisp (run-program port) -
  Current directory: /home/me/prog/next/
 <INFO> [18:23:38] next port.lisp (run-program port) -
  Platform port path: /home/me/prog/next/ports/gtk-webkit/next-gtk-webkit
 <INFO> [18:23:38] next port.lisp (run-program port) -
  Platform port arguments: NIL
 <INFO> [18:23:38] next port.lisp (run-program port) -
  Platform port log file: /home/me/.local/share/next/next-gtk-webkit.log
WARNING:
   Method handler (NEXT::CORE-OBJECT
                   NEXT::PUSH-INPUT-EVENT) signaled an error: Missing right bracket to close character c
lass. at position 3 in string ".*[.*".
```

**Comment by jmercouris** [comment]
Hi, thank you very much for the report! You are exactly correct, it does in fact build regex without escaping the chars. There is a function in utility.lisp called `fuzzy-match`. Do you have any suggestions for how we may fix this? Thanks!

**Comment by Ambrevar** [comment]
Fixed in 96f2ac788ebf20d058ecbf099b24c13be87ff274.
